### PR TITLE
[Merged by Bors] - feat: port CategoryTheory.Types

### PR DESCRIPTION
**Pull request by mattrobball**
---

- [x] depends on: #2292

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#2292
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#2292~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/CategoryTheory/Types.lean [hunk 1]
```diff
@@ -0,0 +1,405 @@
...
+#align category_theory.inv_hom_id_apply CategoryTheory.inv_hom_id_apply
+
+-- Unfortunately without this wrapper we can't use `CategoryTheory` idioms, such as `IsIso f`.
+/-- `asHom f` helps Lean type check a function as a morphism in the category `Type`. -/
+abbrev asHom {α β : Type u} (f : α → β) : α ⟶ β :=
+  f
+#align category_theory.as_hom CategoryTheory.asHom
+
+-- If you don't mind some notation you can use fewer keystrokes:
+/-- Type as \upr in VScode -/
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/Types.lean:92]: VScode automatically gives suggestions about how to type unicode symbols as part of hover tooltips. So let's copy the docstring from above
```suggestion
@[inherit_doc]
```

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**